src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/client/AddressServiceClientServiceImpleTest.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.util.MultiValueMap;
+import uk.gov.ons.ctp.common.FixtureHelper;
+import uk.gov.ons.ctp.common.rest.RestClient;
+import uk.gov.ons.ctp.integration.contactcentresvc.config.AddressIndexSettings;
+import uk.gov.ons.ctp.integration.contactcentresvc.config.AppConfig;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.AddressQueryRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.PostcodeQueryRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.service.addressindex.model.AddressIndexSearchResultsDTO;
+
+public class AddressServiceClientServiceImpleTest {
+  @Mock AppConfig appConfig = new AppConfig();
+
+  @Mock RestClient restClient;
+
+  @InjectMocks
+  AddressServiceClientServiceImpl addressClientService = new AddressServiceClientServiceImpl();
+
+  @Captor ArgumentCaptor<String> c;
+
+  @Captor ArgumentCaptor<MultiValueMap<String, String>> queryParamsCaptor;
+
+  @Before
+  public void initMocks() {
+    MockitoAnnotations.initMocks(this);
+
+    // Mock the address index settings
+    AddressIndexSettings addressIndexSettings = new AddressIndexSettings();
+    addressIndexSettings.setAddressQueryPath("/addresses");
+    addressIndexSettings.setPostcodeLookupPath("/addresses/postcode");
+    Mockito.when(appConfig.getAddressIndexSettings()).thenReturn(addressIndexSettings);
+  }
+
+  @Test
+  public void testAddressQueryProcessing() throws Exception {
+    // Build results to be returned from search
+    AddressIndexSearchResultsDTO addressIndexResults =
+        FixtureHelper.loadClassFixtures(AddressIndexSearchResultsDTO[].class).get(0);
+    Mockito.when(
+            restClient.getResource(
+                eq("/addresses"), eq(AddressIndexSearchResultsDTO.class), any(), any(), any()))
+        .thenReturn(addressIndexResults);
+
+    // Run the request and sanity check the results. We can't thoroughly check the data as it
+    // is not coming from a fixed test data set
+    AddressQueryRequestDTO request = AddressQueryRequestDTO.create("Michael", 0, 100);
+    AddressIndexSearchResultsDTO results = addressClientService.addressQuery(request);
+    assertEquals("39", results.getDataVersion());
+    assertEquals(4, results.getResponse().getAddresses().size());
+
+    // Verify that the query parameters being passed to AddressIndex are as expected
+    Mockito.verify(restClient).getResource(any(), any(), any(), queryParamsCaptor.capture(), any());
+    MultiValueMap<String, String> queryParams = queryParamsCaptor.getValue();
+    assertEquals("[Michael]", queryParams.get("input").toString());
+    assertEquals("[0]", queryParams.get("offset").toString());
+    assertEquals("[100]", queryParams.get("limit").toString());
+    assertEquals(3, queryParams.keySet().size());
+  }
+
+  @Test
+  public void testPostcodeQueryProcessing() throws Exception {
+    // Build results to be returned from search
+    AddressIndexSearchResultsDTO addressIndexResults =
```

Comment:
is it one result or many?

Reply:
I've renamed the variable to 'resultsFromAddressIndex'.

Not sure exactly what you are asking, but the resource file does contains a json structure with 39 addresses that we are pretending that the address Index service has returned.